File: backend/app/schemas/review.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class ReviewScheduleBase(BaseModel):
    item_type: str
    next_review_date: datetime
    review_count: int = 0
    ease_factor: float = Field(default=2.5, ge=1.3, le=5.0)
    interval_days: int = 0
    status: str = "learning"


class ReviewSubmit(BaseModel):
    item_type: str
    item_id: int
    score: int = Field(ge=0, le=5)  # User self-rating 0-5


class ReviewResult(BaseModel):
    next_review_date: datetime
    interval_days: int
    review_count: int
    status: str


class ReviewItemResponse(BaseModel):
    id: int
    item_type: str
    item_id: int
    word: Optional[str] = None  # Word text if item_type is 'word'
    sentence: Optional[str] = None  # Sentence text if item_type is 'sentence'
    meaning: Optional[str] = None
    review_count: int
    interval_days: int
    next_review_date: datetime
    status: str

    class Config:
        from_attributes = True


class TodayReviewResponse(BaseModel):
    due_count: int
    new_count: int
    review_items: List[ReviewItemResponse]


class ReviewStats(BaseModel):
    total_words: int
    total_sentences: int
    mastered_count: int
    learning_count: int
    review_count: int
    today_reviews: int
    tomorrow_reviews: int

File: frontend/src/types/review.ts
export interface ReviewSubmit {
  item_type: string
  item_id: number
  score: number  // 0-5
}

export interface ReviewResult {
  next_review_date: string
  interval_days: number
  review_count: number
  status: string
}

export interface ReviewItem {
  id: number
  item_type: string
  item_id: number
  word?: string
  sentence?: string
  meaning?: string
  review_count: number
  interval_days: number
  next_review_date: string
  status: string
}

export interface TodayReviewResponse {
  due_count: number
  new_count: number
  review_items: ReviewItem[]
}

export interface ReviewStats {
  total_words: number
  total_sentences: number
  mastered_count: number
  learning_count: number
  review_count: number
  today_reviews: number
  tomorrow_reviews: number
}